tests/framework/extensions/nodes/node_status.go

```diff
@@ -0,0 +1,47 @@
+package nodes
+
+import (
+	"time"
+
+	"github.com/rancher/norman/types"
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+const (
+	active = "active"
+)
+
+// IsNodeReady is a helper method that will loop and check if the node is ready in the RKE1 cluster.
+// It will return an error if the node is not ready after set amount of time.
+func IsNodeReady(client *rancher.Client, ClusterID string) error {
```

Comment:
Another question about the NodeReady, function that I missed. Should we use the Steve client instead to do the Node check? If this is going to be used for all the different cluster types, I think we should do that. Correct me if I'm but I believe rancher use the Steve(v1) API for the nodes. This could then settle the issue of having two `IsNodeReady` funcs, one for rke1 node pools, and on for all cluster nodes.

Reply:
Oh that would be great if that's the case, I'll look into that!

Reply:
Thank you!

Reply:
I can actually use Management client for both. I updated `client.Management.Node.List` to `client.Management.Node.ListAll`. Verified that RKE1/RKE2/K3s that this is working as expected, so I can just remove the duplicate function in `nodepools.go` now.

Reply:
Resolved.

Reply:
Awesome, but wanted to make sure on one thing. Does the UI use v3 or v1 when getting the nodes?

Reply:
v3 I believe. At least when I've been doing testing, v3 is in all of these tests and the web requests seemed to utilize v3 as well.

Reply:
Awesome!!! Thank you just wanted to make sure